### PR TITLE
Updated to latest cli-kit for better banner handling

### PR DESCRIPTION
**Pull request by cb1kenobi**
Update appcd and its commands to use the latest cli-kit API.

Replaced `del` with `fs-extra`: https://jira.appcelerator.org/browse/DAEMON-258.

Updated npm deps.

### CODE REVIEW

File: packages/appcd/src/main.js [hunk 1]
```diff
@@ -7,6 +7,16 @@ import cli from './cli';
...
-		process.exit(err.exitCode || 1);
+		const exitCode = err.exitCode || 1;
+
+		if (err.json) {
+			console.log(JSON.stringify({
+				code: exitCode,
+				result: err.toString()
+			}, null, 2));
+		} else {
+			console.error(err.message);
```
Comment (ewanharris) [inline packages/appcd/src/main.js]: Kinda not related. 

I ran into some problems with the standard of logging only `err.message` and tracking the error down, could we maybe use snooplogg to log the full error so when this hits users machines we're not having them editing the source to give us stacks

File: packages/appcd/src/commands/logcat.js [hunk 2]
```diff
@@ -18,54 +18,55 @@ const cmd = {
...
-
-		let filter = '*';
-		const filters = arrayify(argv.filters);
-		if (filters.length) {
-			if (filters.every(a => a[0] === '-')) {
-				// if every filter arg is a negation, then that means there are no allowed
-				// namespace and want we really want is everything except said filters
-				filter = `* ${filters.join(' ')}`;
+			let formatter;
+			if (argv.colors) {
```
Comment (ewanharris) [inline packages/appcd/src/commands/logcat.js]: Should be `argv.color` now I think?

File: packages/appcd-gulp/src/babel.js [hunk 3]
```diff
@@ -1,4 +1,8 @@
-{
+const fs     = require('fs');
```
Comment (ewanharris) [inline packages/appcd-gulp/src/babel.js]: The babel changes here broke running tests. I have changes locally that make this export the profiles as profiles and the function it does currently as `init`, so the changes you made would be `require('./babel.js').init(opts)` and the test require becomes `const conf = require('./babel.js').profiles[process.env.APPCD_BABEL_CONF || 'node8'];`, shall I push or do you wish to fix yourself?

Comment (cb1kenobi) [reply packages/appcd-gulp/src/babel.js]: Fixed.

File: packages/appcd-core/CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,16 @@
-# v1.1.4
+# v2.0.0
```
Comment (ewanharris) [inline packages/appcd-core/CHANGELOG.md]: Missing the version bump in package.json? I guess we'd also need to update the plugins to support this version too?

File: packages/appcd-util/src/util.js [hunk 5]
```diff
@@ -427,6 +427,74 @@ export function mutex(name, callback) {
 		}));
 }
 
+/**
+ * Tries to resolve the operating system name and version.
+ *
+ * @returns {Object}
+ */
+export function osInfo() {
```
Comment (ewanharris) [inline packages/appcd-util/src/util.js]: I think we do something similar maybe in the system-info plugin. Should we maybe move that to use this function at some point?

Comment (cb1kenobi) [reply packages/appcd-util/src/util.js]: That's the idea. I'll update the system-info plugin to this new `osInfo()` in appcd-util once we publish it. :)

File: packages/appcd-gulp/package.json [hunk 6]
```diff
@@ -13,39 +13,40 @@
     "@babel/plugin-transform-async-to-generator": "latest",
     "@babel/polyfill": "latest",
     "@babel/register": "latest",
```
Comment (ewanharris) [inline packages/appcd-gulp/package.json]: Now that babel 7 is released, any objections to replacing latest with ^7.0.0 here?

Comment (cb1kenobi) [reply packages/appcd-gulp/package.json]: Yes, I will do this! Thanks!

**Comment by ewanharris** [comment]
@cb1kenobi Pushed a commit to fix the winreg issue

**Pull request merged**